### DISCUSSION TITLE
Make bullet shells and other low-utility entities eventually despawn

### DISCUSSION THREAD

**Discussion by moonheart08** [space-wizards/space-station-14#3973] [Ideaguys]
If left to lie around, they can really gum up the gears of the server as far as I can tell. Either this, or mark them "special" in some way that prevents the server from doing any extra calculations on them.

As per CakeQ this should be a config option, as, for example, servers like CM could care less about keeping bullet casings all over the floor (imo), while heavily RP oriented servers would want to keep them around for detective work.

**Comment by CakeQ** [comment]
In the grand scheme of things you want this to be tied under a configuration as things like bullet casings will be used to collect evidence and solve crimes by detectives in a typical ss13/14 game.

**Comment by ghost** [comment]
I don't think having bullet casings should be an issue as long as netcode is optimized for it

**Comment by PrPleGoo** [comment]
Casings can often be collected for metal by shoving them into an autolathe.

**Comment by ShadowCommander** [comment]
This is really only going to be a problem if the game mode includes a lot of guns. Most SS13 rounds don't have many rounds fired from what I can tell, so I'd keep it as non-despawning or as CakeQ said it should be a config setting.

**Comment by TheGandalf85** [comment]
Agree with the configuration options for removing bullet casings after a certain amount of time.

While they are extremely handy for a typical SS13 game and kinda add to the complexity that comes with the game itself, the times that casings are used that are warranted to keep them forever with whatever code time they might bring _can_ be less beneficial then outright making them temporary even with a long timer. However, if optimized properly, they should stick around completely by default.

We can go a bit further and add temporary flags to any entity which would get cleaned up on a similar config timer if they aren't interacted with in a while. In general, this could be an option for games/servers that truly need less of the clutter around.
